File: src/mongoweb/controller/role.js
//
db.user.aggregate(
    [
        {
            $group: {
                _id: "$roleName",
                roleNameShow: {$push: "$roleNameShow"},
                count: {$sum: 1},
                usernames: {$push: "$username"},
            }
        }
    ]
).pretty()


db.user.aggregate(
    [
        {
            $group: {
                _id: {
                    roleName: "$roleName",
                    roleNameShow: "$roleNameShow",
                },
                count: {$sum: 1},
                usernames: {$push: "$username"},
            }
        }
    ]
).pretty()


db.user.update(
    {'roleName': 'user'},
    {$set: {'roleNameShow': '用户'}},
    {multi: true}
)

db.user.update(
    {'roleName': 'admin'},
    {$set: {'roleNameShow': 'admin'}},
    {multi: true}
)


db.user.find({}, {
    "roleName": 1,
    "roleNameShow": 1,
}).pretty()



//报错，那个逗号不能加，
//有的地方多个逗号没事，这个地方多逗号报错，
db.user.find(
    {"roleName": "user",},
    {"roleNameShow": 1,},
).pretty()


db.user.find(
    {"roleName": "user",},
    {"roleNameShow": 1,}
).pretty()

File: src/mongoweb/model/role.go
package model

type Role struct {
	RoleName     string `json:"roleName" bson:"roleName"`         //
	RoleNameShow string `json:"roleNameShow" bson:"roleNameShow"` //
}

//type RoleResult struct {
//	RoleName      string   `json:"roleName" bson:"_id"`           //
//	RoleNameShows []string `json:"roleNameShows" bson:"roleNameShows"` //
//	Count         int      `json:"count" bson:"count"`                 //
//	Usernames     []string `json:"usernames" bson:"usernames"`         //
//}

type RoleResult struct {
	Role      Role     `json:"role" bson:"_id"`            //
	Count     int      `json:"count" bson:"count"`         //
	Usernames []string `json:"usernames" bson:"usernames"` //
}
